openslides/motions/migrations/0003_motion_comments.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+from ..config_variables import get_config_variables
+
+
+def change_motions_comments(apps, schema_editor):
+    """
+    Index the comments fields in the config. Changing from an array to a dict with the
+    ids as keys. CHange all motions from an array for comments to a dict with the comments
+    field id as key to link motion comments and comments fields.
+    """
+    # We get the model from the versioned app registry;
+    # if we directly import it, it will be the wrong version.
+    ConfigStore = apps.get_model('core', 'ConfigStore')
+    Motion = apps.get_model('motions', 'Motion')
+
+    try:
+        config_comments_fields = ConfigStore.objects.get(key='motions_comments').value
+    except ConfigStore.DoesNotExist:
+        config_comments_fields = get_motions_comments_default()
+
+    comments_fields = {}
+    index = 0
+    for field in config_comments_fields:
+        index += 1
+        comments_fields[index] = field
+
+    try:
+        db_value = ConfigStore.objects.get(key='motions_comments')
+    except ConfigStore.DoesNotExist:
+        db_value = ConfigStore(key='motions_comments')
+    db_value.value = comments_fields
+    db_value.save()
+
+    max_index = index
+
+    for motion in Motion.objects.all():
+        comments = {}
+        index = 0
+        for comment in motion.comments or []:
+            index += 1
+            if index > max_index:
+                break
```

Comment:
Ah this is tricky. "Hidden" comments are now deleted. Are you sure?

Reply:
Yes, because there does not exist a commebts field for them. If you start OS, you wouldn't see the comments.

Reply:
Ok for me.